src/handlers/shared/adapter-middleware.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  SALEOR_API_URL_HEADER,
+  SALEOR_AUTHORIZATION_BEARER_HEADER,
+  SALEOR_EVENT_HEADER,
+  SALEOR_SCHEMA_VERSION,
+  SALEOR_SIGNATURE_HEADER,
+} from "@/const";
+
+import { HTTPMethod, PlatformAdapterInterface } from "./generic-adapter-use-case-types";
+
+export class PlatformAdapterMiddleware<T> {
```

Comment:
Maybe better name would be to mention what it does (validating requests), not that it uses adapter for it?